src/Umbraco.Cms.Api.Management/Controllers/Package/PackageControllerBase.cs

```diff
@@ -0,0 +1,26 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Api.Common.Builders;
+using Umbraco.Cms.Api.Management.Routing;
+using Umbraco.Cms.Core.Services.OperationStatus;
+
+namespace Umbraco.Cms.Api.Management.Controllers.Package;
+
+[ApiController]
+[VersionedApiBackOfficeRoute("package")]
+[ApiExplorerSettings(GroupName = "Package")]
+[ApiVersion("1.0")]
+public abstract class PackageControllerBase : ManagementApiControllerBase
+{
+    protected IActionResult PackageMigrationOperationStatusResult(PackageMigrationOperationStatus status) =>
+        status switch
+        {
+            PackageMigrationOperationStatus.NotFound => NotFound("No migration plans were found for that package"),
+            PackageMigrationOperationStatus.CancelledByFailedMigration => Conflict(new ProblemDetailsBuilder()
+                .WithTitle("Package migration failed")
+                .WithDetail("Check log for full details about the failed migration.")
+                .WithStatus(StatusCodes.Status409Conflict)
```

Comment:
I'm wondering if this should be a 500 instead of 409, since if a package migration failed that's really an internal server error in reality?

Reply:
We discussed this with @bergmania, so that's what we agreed on 😁